src/context/notesContext.js

```diff
@@ -0,0 +1,199 @@
+          type: "Error",
+          msg: "Unable to Delete Note",
+        });
+        console.error(error);
+      }
+    };
+    const updateNote = async (notes, content) => {
+      const note = {
+        ...notes,
+        description: parse(content).props.children
+      };
+      console.log(note)
+      notesDispatch({ type: "UPDATE_NOTE", payload:note});
+      // try {
+      //   const { status, data } = await privateInstance({
+      //     method: "post",
+      //     url: `/notes/${id}`,
+      //   });
+      //   if (status === 200) {
+      //     notesDispatch({ type: "UPDATE_NOTE", payload: id });
+      //     return data.notes;
+      //   }
+      // } catch (error) {
+      //   addToast({
+      //     type: "Error",
+      //     msg: "Unable to Update Note",
+      //   });
+      //   console.error(error);
+      // }
+    };
+
+    const getArchiveNotes = async () => {
+      try {
+        const { status, data } = await privateInstance.get("/archives");
+        if (status === 200)
+          archiveNotesDispatch({
+            type: "SET_ARCHIVE_DATA",
+            payload: data.archives,
+          });
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Fetch Data from API",
+        });
+        console.error(error);
+      }
+    };
+    const addNoteToArchive = async (note) => {
+      console.log(note);
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      archiveNotesDispatch({ type: "REMOVE_NOTE", payload: note._id });
+      try {
+        const { status, data } = await privateInstance({
+          method: "post",
+          url: `/notes/archives/${note._id}`,
+          data: {
+            note,
+          },
+        });
+        if (status === 201) {
+          archiveNotesDispatch({ type: "ADD_ARCHIVE_NOTE", payload: note });
+          return data;
+        }
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Add Note to Archive",
+        });
+        console.error(error);
+      }
+    };   
+    const restoreNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note._id });
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      try {
+        const { status, data } = await privateInstance({
+          method: "post",
+          url: `/notes/archives/restore/${note._id}`,
+          data: {},
+        });
+        if (status === 201) {
+          archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note });
+          notesDispatch({ type: "ADD_NOTE", payload: note });
+          return data;
+        }
+      } catch (error) {
+        addToast({
+          type: "Error",
+          msg: "Unable to Restore Note from Archive",
+        });
+        console.error(error);
+      }
+
+    }
+    const deleteNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note });
```

Comment:
again dispatched in line `165` ?

Reply:
sorry forgot to comment that code.
Done.